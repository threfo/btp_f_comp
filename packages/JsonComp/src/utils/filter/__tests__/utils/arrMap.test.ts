import arrMapFilter from '../../utils/arrMap'
describe('utils/filter/utils/arrMap.ts', () => {
  it('arrMapFilter', () => {
    expect(
      arrMapFilter({
        ruleMap: {
          external_link: {
            rule: {
              key: 'page_template',
              valueKeyMap: {
                home_page: '/',
                jobs_page: '/job'
              }
            },
            key: 'url'
          },
          name: {
            key: 'label'
          },
          disabled: {}
        },
        data: [
          {
            name: '66666',
            jump_type: 'template_site',
            page_template: 'home_page',
            disabled: false
          },
          {
            name: '2343',
            jump_type: 'template_site',
            page_template: 'jobs_page'
          },
          {
            name: '2343',
            jump_type: 'template_site',
            external_link: 'http://baidu.com'
          }
        ]
      })
    ).toMatchSnapshot()

    expect(
      arrMapFilter({
        ruleMap: {
          jump_link: {
            key: 'href'
          },
          main_copy: {
            key: 'info',
            rule: {
              model: 'join',
              key: 'html',
              otherProps: {
                class:
                  'xl:text-4xl text-3xl whitespace-pre-wrap font-semibold text-blue-300'
              }
            }
          },
          sub_copy: {
            key: 'info',
            rule: {
              model: 'join',
              key: 'html',
              otherProps: {
                class:
                  'xl:text-xl text-lg whitespace-pre-wrap font-semibold text-gray-500'
              }
            }
          },
          description: {
            key: 'info',
            rule: {
              model: 'join',
              key: 'html',
              otherProps: {
                class: 'whitespace-pre-wrap font-semibold text-gray-800'
              }
            }
          }
        },
        data: [
          {
            picture: {
              id: '6217368dedf54d25aff9b050'
            },
            main_copy: '主文案',
            sub_copy: '副文案',
            description: '内容描述',
            jump_link: 'http://www.baidu.com'
          },
          {
            picture: {
              id: '6217368dedf54d25aff9b050'
            },
            main_copy: '主文案',
            description: '内容描述',
            jump_link: 'http://www.baidu.com'
          }
        ]
      })
    ).toMatchSnapshot()
  })
})
