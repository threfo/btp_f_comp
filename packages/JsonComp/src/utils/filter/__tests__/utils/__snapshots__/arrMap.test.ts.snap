// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/filter/utils/arrMap.ts arrMapFilter 1`] = `
Array [
  Object {
    "disabled": false,
    "label": "66666",
    "url": "/",
  },
  Object {
    "disabled": undefined,
    "label": "2343",
    "url": "/job",
  },
  Object {
    "disabled": undefined,
    "label": "2343",
    "url": "http://baidu.com",
  },
]
`;

exports[`utils/filter/utils/arrMap.ts arrMapFilter 2`] = `
Array [
  Object {
    "href": "http://www.baidu.com",
    "info": Array [
      Object {
        "class": "xl:text-4xl text-3xl whitespace-pre-wrap font-semibold text-blue-300",
        "html": "主文案",
      },
      Object {
        "class": "xl:text-xl text-lg whitespace-pre-wrap font-semibold text-gray-500",
        "html": "副文案",
      },
      Object {
        "class": "whitespace-pre-wrap font-semibold text-gray-800",
        "html": "内容描述",
      },
    ],
  },
  Object {
    "href": "http://www.baidu.com",
    "info": Array [
      Object {
        "class": "xl:text-4xl text-3xl whitespace-pre-wrap font-semibold text-blue-300",
        "html": "主文案",
      },
      Object {
        "class": "whitespace-pre-wrap font-semibold text-gray-800",
        "html": "内容描述",
      },
    ],
  },
]
`;
